DataProcessing/datax-host/src/main/scala/datax/sink/BlobSinker.scala

```diff
@@ -213,5 +229,37 @@ object BlobSinker extends SinkOperatorFactory {
     )
   }
 
+  private def getStorageAccountName(path:String):String ={
+    val regex = "@([a-zA-Z0-9-_]+).blob.core.windows.net".r
+    regex.findFirstMatchIn(path) match {
+      case Some(partition) => partition.group(1)
+      case None =>  null
+    }
+  }
+
+  /***
+    * set storage account key with a keyvault name for writting to blobs on executer nodes
+    * @param vaultName key vault name to get the key of storage account
+    * @param sa name of the storage account
+    */
+  private def setStorageAccountKeyForExecuterNodes(vaultName: String, sa: String): String = {
```

Comment:
This is duplicate code from HadoopClient.resolveStorageAccount , lets just reuse it.

Reply:
Fixed